05/app.js

```diff
@@ -0,0 +1,53 @@
+function Student(initialName, initialLastName) {
+    this.firstName = initialName;
+    this.lastName = initialLastName;
+    this.grades = {};
+}
+
+Student.prototype.addGrade = function (subject, grade) {
+    if (typeof this.grades[subject] === 'undefined') {
+        this.grades[subject] = [];
+    }
+
+    this.grades[subject].push(grade);
+}
+
+Student.prototype.getAvgGrade = function (subject) {
+    const subjectGrades = this.grades[subject];
+
+    // Przy wyliczeniu średniej ze wszystkich przedmiotów posiłkowałem się podobnym pytaniem na slacku
+    const grades = Array.from(Object.values(this.grades)).flat();
```

Comment:
To rozwiązanie jest bardziej PRO :) moim zdaniem lepiej napisać coś co się rozumie.
Jeśli wiesz co się tutaj dzieje to ok, jeśli nie to starałbym się to przepisać np.
1. Utworzyć pustą tablicę.
2. Wykorzystać pętle for in, aby iterować po przedmiotach, 
3. a pętle for, aby z przedmiotu pobrać konkretną oceną i dodać ją do tablicy z punktu 1

Wychodzi na to samo :)